src/sparseml/modifiers/quantization/pytorch.py

```diff
@@ -73,6 +74,9 @@ def __init__(self, **kwargs):
 
     def on_initialize_structure(self, state: State, **kwargs):
         module = state.model.model
+        # before the structure is modified to support quantization,
+        # we need to potentially modify the model architecture
+        module = modify_model(module)
```

Comment:
Has this been tested with FSDP?? I think that is out of scope of this PR but would be good to know if it works or not

Reply:
Also, do you think we need this on line 95 too? That is where we setup for quantization when the input model hasn't been quantized set

Reply:
@Satrat could we write a unit test that covers the FSDP scenario? I am confused, what do you mean by line 95?

Reply:
Oh I see: `self._enable_module_qat(module)`. It seems like it. When would the model go through line 95 and not `on_initialize_structure` method?

Reply:
Since FSDP requires multiple GPUs we don't have any integration tests set up for it. Theres documentation on generating an FSDP config here: https://github.com/neuralmagic/sparseml/blob/main/integrations/huggingface-transformers/finetuning/configure_fsdp.md, and you can use that config for the `tests/sparseml/transformers/obcq/test_obcq.py:test_obcq_tinystories` integration test with the `quant.yaml` recipe parameter

Reply:
As for the line 95 comment, `on_initialize_structure` is for initializing a recipe that has been previously applied to a model. So it would be used in the context of reloading a quantized model. `on_initialize` is called when you're quantizing a model for the first time